pkg/fileio/file_io.go

```diff
@@ -107,7 +107,17 @@ func CopyFiles(src, dest, ext string, copySubDir bool) error {
 				continue
 			}
 
-			err := CopyFile(sourcePath, destPath, NonExecutablePerms)
+			perms := NonExecutablePerms
+			if maintainPerms {
```

Comment:
I'm not quite convinced - why do we need this param?

Reply:
So my idea here is that for things that we use internally, we don't need to maintain the original permissions, but for things that the user provides to be on the final image that they manage and we don’t, it should be the original permissions they specified. To this, I thought having that parameter would be a clean solution to allow us to easily specify how we want to use the function wherever we need to. The usage itself though is up for discussion.

Reply:
I still think this is unnecessary complexity - none of the files that we transfer are executable in the first place.

Reply:
Isn't this for the custom files that a user can place in a tree?

Reply:
Another consideration here is that we'll eventually need to maintain SELinux contexts. In that light, I think the approach of "always maintain whatever is on the filesystem" is probably a better approach. If I try to step back as a completely new user, I think that's what I'd expect the behavior to be.